crates/services/consensus_module/poa/src/service_test.rs

```diff
@@ -166,23 +170,46 @@ impl TestContextBuilder {
 
         let service = new_service(
             &BlockHeader::new_block(BlockHeight::from(1u32), Tai64::now()),
-            config,
+            config.clone(),
             txpool,
             producer,
             importer,
             p2p_port,
+            FakeBlockSigner { succeeds: true },
             predefined_blocks,
         );
         service.start().unwrap();
         TestContext { service }
     }
 }
 
+struct FakeBlockSigner {
+    succeeds: bool,
+}
+
+#[async_trait::async_trait]
+impl BlockSigner for FakeBlockSigner {
+    async fn seal_block(&self, block: &Block) -> anyhow::Result<Consensus> {
+        if self.succeeds {
+            SignMode::Key(Secret::new(default_consensus_dev_key().into()))
+                .seal_block(block)
+                .await
+        } else {
+            Err(anyhow::anyhow!("failed to sign block"))
+        }
+    }
+
+    fn is_available(&self) -> bool {
```

Comment:
I think we should have some tests for this method.

Reply:
https://github.com/FuelLabs/fuel-core/issues/2101